packages/mattermost-redux/src/client/client4.ts

```diff
@@ -3737,6 +3737,10 @@ export default class Client4 {
         const response = await fetch(url, this.getOptions(options));
         const headers = parseAndMergeNestedHeaders(response.headers);
 
+        if (this.getToken() === undefined && headers.has('Token')) {
```

Comment:
Also, maybe this can be done in response to just the login request [like we used to do before](https://github.com/mattermost/mattermost-redux/blame/a10b6c7f67a5163c15d029defbcd9bccf0eb0412/src/client/client4.js#L450)? It shouldn't make much difference overall, but I think it's nice to keep it with the login method itself.